Add documentation of Azure Service Principal to the README (#10)
databricks/run-notebook: README.md

and [generate an API token](https://docs.databricks.com/dev-tools/api/latest/token-management.html#operation/create-obo-token) on its behalf.

## Azure
For security reasons, we recommend using a Databricks service principal AAD token.

> You can:

Reviewer: ```suggestion

### Create an Azure Service Principal
You can:
...
### Use the Service Principal in your GitHub Workflow
* Add the following snippet to the start of your GitHub workflow. This will create a new AAD token and save its value in the `DATABRICKS_TOKEN` environment variable for use in subsequent steps
(include YAML snippet here or link to one of the snippets below. Snippet can contain code comments explaining the resource ID).
```

I wonder if it'd help for clarity to separate the steps for creating the Azure SP and using it in the workflow. Also, I think this section is probably just best demonstrated through a code example

Author: Sounds Good! done

* Get the application id of your new service principal by running `az ad sp show --id <clientId from previous command output>`, using
  the `clientId` field from the JSON output of the previous step.
* [Add your service principal](https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/scim/scim-sp#add-service-principal) to your workspace. Use the
  `appId` output field of the previous step as the `applicationId` of the service principal in the `add-service-principal` payload.
* At the start of your Workflow, use the [Azure/login Action](https://github.com/Azure/login) to authenticate to Azure
  as your service principal, passing `${{ secrets.AZURE_CREDENTIALS }}` from the previous step.
* In a following workflow step, run `echo "DATABRICKS_TOKEN=$(az account get-access-token --resource=2ff814a6-3304-4ab8-85cb-cd0e6f879c1d --query accessToken -o tsv)" >> $GITHUB_ENV`
  to create an AD token on behalf of the service principal and assign its value to the `DATABRICKS_TOKEN` environment variable. The
  `databricks/run-notebook` Action will automatically detect and use the value in `DATABRICKS_TOKEN` to authenticate

>   to Databricks. Note: We pass into the `get-access-token` cli command the programmatic Azure Databricks id as `resourceId`.

Author: Let me know if you think the 2 added `Note:` are clear

* Get the application id of your new service principal by running `az ad sp show --id <clientId from previous command output>`, using
  the `clientId` field from the JSON output of the previous step.
* [Add your service principal](https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/scim/scim-sp#add-service-principal) to your workspace. Use the
  `appId` output field of the previous step as the `applicationId` of the service principal in the `add-service-principal` payload.
* **Note**: The generated Azure token has a default life span of **60 minutes**.
  If you expect your Databricks notebook to take longer than 60 minutes to finish executing, then you must create a [token lifetime policy](https://docs.microsoft.com/en-us/azure/active-directory/develop/configure-token-lifetimes)
  and attach it to your service principal.

### Use the Service Principal in your GitHub Workflow
* Add the following steps to the start of your GitHub workflow.
  This will create a new AAD token and save its value in the `DATABRICKS_TOKEN`
  environment variable for use in subsequent steps.

  ```yaml

>   # Obtain an AAD token and use it to run the notebook on Databricks

Reviewer: ```suggestion
```

Reviewer: Since we're not running the notebook in this snippet

  - name: Log into Azure
    uses: Azure/login@v1
    with:
      creds: ${{ secrets.AZURE_CREDENTIALS }}

>   # Get an AAD token for the SP using the programmatic ID for Azure Databricks as resourceId.

Reviewer: ```suggestion
      # Get an AAD token for the service principal,
      # and store it in the DATABRICKS_TOKEN environment variable for use in subsequent steps.
      # We set the `resource` parameter to the programmatic ID for Azure Databricks.
      # See https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/aad/service-prin-aad-token#--get-an-azure-ad-access-token for details.
```

>   # Get an AAD token for the SP using the programmatic ID for Azure Databricks as resourceId.

Reviewer: ```suggestion
```

We may be able to delete this since it's covered in the text above

  `databricks/run-notebook` Action will automatically detect and use the value in `DATABRICKS_TOKEN` to authenticate
  to Databricks. Note: We pass into the `get-access-token` cli command the programmatic Azure Databricks id as `resourceId`.
  This is necessary for authentication with your Azure Databricks workspace. Also note that the generated Azure token has a default life span of 60 minutes.

>   If you expect your Databricks notebook to take longer than 60 minutes to finish executing, then you must create a [token lifetime policy](https://docs.microsoft.com/en-us/azure/active-directory/develop/configure-token-lifetimes)

Reviewer: We may want to call this out more visibly (higher up e.g. when we start discussing how to use the Service Principal in a GitHub Workflow) since it's a pretty big gotcha

Author: Moved as a note in the `Create an Azure Service Principal` section before the `Use the Service Principal in your GitHub Workflow` section

    steps:
      - name: Checkout repo
        uses: actions/checkout@v2
      # Obtain an AAD token and use it to run the notebook on Databricks
      # Note: If running on AWS or GCP, you can directly pass your service principal
      # token via the databricks-host input instead
      - name: Log into Azure
        uses: Azure/login@v1
        with:
          creds: ${{ secrets.AZURE_CREDENTIALS }}

>       # Get an AAD token for the SP using the programmatic ID for Azure Databricks as resourceId.

Reviewer: ```suggestion
      # Get an AAD token for the service principal, and store it in the DATABRICKS_TOKEN environment variable for use in subsequent steps. We set the `resource` parameter to the programmatic ID for Azure Databricks. See https://docs.microsoft.com/en-us/azure/databricks/dev-tools/api/latest/aad/service-prin-aad-token#--get-an-azure-ad-access-token for details.
```

Reviewer: Would be nice to add a doc link to explain the magic string :) so people can confirm it's legit

Author: Done!

>       # Then setting the returned accessToken to env.DATABRICKS_TOKEN.

Reviewer: ```suggestion
```

          access-control-list-json: >
            [
              {

>                 "users": "Can View"

Author: The example we had for `access-control-list-json` did not follow expected input format for: https://docs.databricks.com/dev-tools/api/latest/jobs.html#operation/JobsRunsSubmit
